Sidebar style updates

stevedya opened the pull request:
Addresses #7919 styling of sidebar to match ui redesign colour scheme. 
Colour changes to the page explorer portion of the sidebar are temporary to avoid major colour clashes until a new UI design for the page explorer is ready. 

_This doesn't address #7918 for interaction updates_

- [x] Background colour change to wagtail's new primary colour
- [x] Wagtail bird animation tweak and svg refactor
- [x] Update top “expand” icon to match designs
- [x] Move search icon to the left when sidebar is expanded
- [x] Wagtail logo size decrease to circular when sidebar is in slim (collapsed) mode
- [x] Change profile image to be circular
- [x] Styling of account and logout links to match design
- [x] Match colours of subnavigation backgrounds to new sidebar colour scheme 
- [x] Ensure colour contrast still meets WCAG 2.1 standard 


### Screenshots
 **Nested subnav** 

<img width="402" alt="Screen Shot 2022-03-16 at 9 45 28 AM" src="https://user-images.githubusercontent.com/25041665/158573278-273b0634-6712-4228-8b2c-7b5c04c0b2e6.png">

**Logo on hover**

<img width="201" alt="Screen Shot 2022-03-16 at 9 45 19 AM" src="https://user-images.githubusercontent.com/25041665/158573653-7f8e8992-a38c-4c3d-bc53-085d48d9cedc.png">


**With page explorer**

<img width="682" alt="Screen Shot 2022-03-16 at 9 44 54 AM" src="https://user-images.githubusercontent.com/25041665/158573283-4724d5e6-8663-4c15-bfa3-6cb63f0d92c9.png">


### Browsers tested on

- Chrome 99
- Safari 15

thibaudcolas commented on client/src/components/PageExplorer/PageExplorer.scss:
```diff
@@ -83,7 +74,7 @@ $explorer-header-horizontal-padding: 10px;
 
   .icon {
-    color: $c-page-explorer-secondary;
+    @apply w-text-white/50;
```
With this change, we’d be using different icon colors between the header and the individual items inside the explorer. Is that intentional?

stevedya replied:
This is intentional, but I'll do another pass and make sure they all match this colour, great catch!

thibaudcolas commented on client/src/components/PublicationStatus/PublicationStatus.scss:
```diff
@@ -1,6 +1,3 @@
-  text-transform: uppercase;
-  letter-spacing: 0.03rem;
-  font-size: 10px;
+  @apply w-bg-black/50 w-uppercase w-tracking-tight w-text-[10px] w-text-white/85;
```
Same as above – I’d prefer we keep the font-size done with CSS, so it stands out more that it really shouldn’t be in pixels, and shouldn’t be so small.

thibaudcolas reviewed (commented):
Partial code review – will keep going tomorrow!

thibaudcolas commented on client/src/components/Sidebar/modules/MainMenu.tsx:
```diff
@@ -202,21 +203,38 @@ export const Menu: React.FunctionComponent<MenuProps> = ({
           type="button"
         >
-          <div className="avatar square avatar-on-dark">
+          <div className="avatar avatar-on-dark !w-w-[28px] !w-h-[28px]">
```
I would have assumed we’d display avatars in consistent sizes elsewhere in the UI, so since there is no `avatar` component in templates / React currently, we should probably do this with a `--small` BEM modifier here?

stevedya replied:
So there is apparently a .small class here https://github.com/wagtail/wagtail/blob/main/client/scss/objects/_avatar.scss#L27

However it's used elsewhere in the admin, so if I was to update it, it would break that other part of the admin (another argument for utility styling and tailwind). 

I'd vote to keep these arbitrary values here because this specific account icon is arbitrary sized itself, and only used here, so to me doesn't seem like much value to entangle the sizing here with more BEM. 

Let me know if I should update `.small` or maybe convert it to `--small` (and do a bunch of extra testing on the other spots its used)? 
or 
If it is okay as is?

thibaudcolas commented on client/src/components/Sidebar/modules/MainMenu.tsx:
```diff
@@ -202,21 +203,38 @@ export const Menu: React.FunctionComponent<MenuProps> = ({
           <div className="sidebar-footer__account-toggle">
-            <div className="sidebar-footer__account-label">{user.name}</div>
+            <div className="sidebar-footer__account-label w-label-3 w-text-white">
+              {user.name}
+            </div>
             <Icon
-              className="sidebar-footer__account-icon"
+              className="w-w-4 w-h-4 w-text-white"
```
Considering it seems the text is white for almost everything in here, if we’re refactoring the code, I’d prefer we set `w-text-white` on the whole sidebar and then use `inherit` in places like this.

thibaudcolas commented on client/src/components/Sidebar/modules/Search.scss:
```diff
@@ -1,16 +1,10 @@
 // stylelint-disable declaration-no-important
 .sidebar-search {
+  @apply w-z-20 w-relative w-box-border w-flex w-items-center w-flex-row w-h-[42px] w-px-5;
```
Can you explain why this needs a `z-index`? I can’t see a difference either way from my testing. If it needs to stay this would warrant a comment.

stevedya replied:
Ahh yes, this was because the tail of the logo was overlapping the search, however I was able to fix this now by adding overflow hidden to the logo until its hovered 🙌

thibaudcolas commented on client/src/components/Sidebar/modules/Search.scss:
```diff
@@ -27,8 +22,6 @@
     inset-inline-start: 0;
     top: 0;
     // Need !important to override body.ready class
-    transition: background-color $menu-transition-duration ease,
-      opacity $menu-transition-duration ease !important;
```
We should probably remove that comment too if we’re removing this.

thibaudcolas commented on client/src/components/Sidebar/modules/Search.tsx:
```diff
@@ -8,6 +8,7 @@ interface SearchInputProps {
   expandingOrCollapsing: boolean;
   searchUrl: string;
   strings: Strings;
+
   navigate(url: string): void;
```
```suggestion
  navigate(url: string): void;
```

thibaudcolas reviewed (changes requested):
The end result looks great @stevedya! There are a few issues with the implementation, and a few more refactorings to do, but we’re not far off. One thing I feel very uncertain about is the move of the "Hamburger" button for mobile from the left to the right. The current `main` implementation is very hacky, so if we do change it, we have to fix all those hacks.

As far as the UI, I could spot the following issues:

- On mobile viewports, the spacing around the account toggle is missing
- On mobile viewports, the hamburger button isn’t aligned with the close button
- On mobile viewports, the space for the "close" button is still there even though the button has been moved to the right side of the screen
- I can’t see the "Close" button on mobile on some pages – it’s there on `/admin/`, but not on `/admin/pages/60/`.
- On pages like /admin/pages/60/edit/, we’ve moved the hamburger button to the right, but the extra space needed to the left hasn’t been removed
- On a 320px viewport, the page explorer scrolls horizontally due not taking over the whole screen anymore. I think we’d want no horizontal scrolling
- In sub-menus, the menu icons don’t seem to have enough contrast against their background. We’d need a minimum of 3:1. Looks like this is because the icon has an opacity of `0.15`, on top of us changing the color to a transparent white.

And more minor things, where I might have it wrong:

- I’d expect the locale selector in the page explorer should keep having the same background as the rest of the explorer header
- I’d expect the colors of the icons in the page explorer should be more consistent

Summary GIF:

![sidebar-styles](https://user-images.githubusercontent.com/877585/158814989-50530b51-a6fe-424e-af5d-e66c85531764.gif)

stevedya commented on client/src/components/PageExplorer/PageExplorer.scss:
```diff
@@ -10,10 +10,10 @@ $menu-footer-height: 50px;
   max-width: 485px;
   width: 90vw;
   height: 100vh;
-  background: $c-page-explorer-bg;
```
Yep good Idea, this will no longer be needed

stevedya commented on client/src/components/PageExplorer/PageExplorerItem.scss:
```diff
@@ -31,9 +29,9 @@
+  @apply w-text-white/85;
   width: 2em;
   height: 2em;
-  color: $c-page-explorer-secondary;
```
I think they should be the same but i missed it in the header. I'll work on fixing this 👀

stevedya commented on client/src/components/PageExplorer/PageExplorerItem.scss:
```diff
@@ -87,7 +83,6 @@
+  @apply w-flex w-gap-2 w-text-white/85 w-text-[12px];
   margin-inline-start: 0.5rem;
-  color: $c-page-explorer-secondary;
-  font-size: 12px;
```
Sure, i'm fine with that :)

stevedya commented on client/src/components/Sidebar/Sidebar.tsx:
```diff
@@ -19,6 +19,7 @@ export interface ModuleRenderContext {
   expandingOrCollapsing: boolean;
   currentPath: string;
   strings: Strings;
+
   navigate(url: string): Promise<void>;
```
haha oooops

stevedya commented on client/src/components/Sidebar/Sidebar.tsx:
```diff
@@ -183,17 +184,18 @@ export const Sidebar: React.FunctionComponent<SidebarProps> = ({
+            <Icon
+              name="expand-right"
+              className={`w-transition motion-reduce:w-transition-none
+                ${!collapsed ? '-w-rotate-180 w-w-6 w-h-6 w-text-white' : ''}
```
Hmm good catch

stevedya commented on client/src/tokens/colors.js:
```diff
@@ -116,6 +116,13 @@ const colors = {
+      hex: '#261A4E',
+      bgUtility: 'w-bg-primary-200',
+      textUtility: 'w-text-primary-200',
+      usage: 'Wagtail branding, Panels, Headings, Buttons, Labels',
```
Yep!

stevedya commented on client/tailwind.config.js:
```diff
@@ -68,6 +70,15 @@ module.exports = {
     spacing,
+    extend: {
+      width: {
+        sidebar: '200px',
```
I was trying to make a push further down the tailwind well here haha. I guess we aren't ready for this yet but one day when we want to go full tailwind this would be a good option for this value.

stevedya commented on client/src/components/Sidebar/modules/MainMenu.scss:
```diff
@@ -88,50 +85,16 @@ $avatar-size: 60px;
-    border: 0;
-    overflow: hidden;
-
     @include clearfix;
```
Removed now 🎉

stevedya commented on client/src/components/Sidebar/modules/WagtailBranding.tsx:
```diff
@@ -11,15 +12,14 @@ export interface LogoImages {
 
 interface WagtailBrandingProps {
   homeUrl: string;
-  images: LogoImages;
```
Oh nice idea, I'll clean this up!

stevedya commented on client/src/components/Sidebar/menu/SubMenuItem.scss:
```diff
@@ -30,25 +30,15 @@
-    text-transform: none;
-    text-align: center;
-    color: $color-menu-text;
+    @apply w-min-h-[160px] w-inline-flex w-flex-col w-justify-center w-items-center w-text-center w-font-medium w-text-white/80 w-mb-0 w-mt-[35px] w-normal-case w-text-18;
```
Okay great points! I've adjusted this based on your feedback
- I've used `w-h4` because it's closer to the size we need. (Not sure if this should then be a <h4> or if we just make it a `<span>`.
- I've added a comment explaining what those arbitrary values are for and rearranged order. 
- I've change the text to be white because the contrast helps it look more like a title 
- Removed unused normal-case and text-center classes.

Also I believe there might be new designs in the future that will focus more on these panels as well as the page explorer.

stevedya commented on client/src/components/Sidebar/modules/MainMenu.scss:
```diff
@@ -88,50 +85,16 @@ $avatar-size: 60px;
-    }
-
-    .avatar {
-      width: $avatar-size;
-      height: $avatar-size;
```
Good shout, I was able to revise this a bit further so we don't need absolute positioning now, also removed this.

stevedya commented on client/src/components/Sidebar/modules/MainMenu.scss:
```diff
@@ -88,50 +85,16 @@ $avatar-size: 60px;
-      }
-    }
-
     $icon-size: 1.5em;
```
Refactored a bit further now so this isn't needed now.

stevedya commented on client/src/components/Sidebar/modules/MainMenu.scss:
```diff
@@ -146,18 +109,19 @@ $avatar-size: 60px;
-      overflow: hidden;
-      white-space: nowrap;
-      text-overflow: ellipsis;
+      @apply w-text-left w-overflow-hidden w-whitespace-nowrap w-text-ellipsis w-capitalize;
```
Good call, i've removed this now

stevedya commented on client/src/components/Sidebar/modules/MainMenu.tsx:
```diff
@@ -202,21 +203,38 @@ export const Menu: React.FunctionComponent<MenuProps> = ({
+          md:w-px-5
+          md:w-py-3
+          hover:w-bg-primary
+          hover:w-text-white focus:w-bg-primary focus:w-text-white"
```
I've adjusted these `hover:` variants here now so only the background hover will show. Also with how this is styled now we can keep the padding on every breakpoint so no need for any `breakpoint:` 🎉

stevedya commented:
> The end result looks great @stevedya! There are a few issues with the implementation, and a few more refactorings to do, but we’re not far off.
> 
> As far as the UI, I could spot the following issues:
> 
> * On mobile viewports, the spacing around the account toggle is missing
> * On mobile viewports, the hamburger button isn’t aligned with the close button
> * On mobile viewports, the space for the "close" button is still there even though the button has been moved to the right side of the screen
> * I can’t see the "Close" button on mobile on some pages – it’s there on `/admin/`, but not on `/admin/pages/60/`.
> * On pages like /admin/pages/60/edit/, we’ve moved the hamburger button to the right, but the extra space needed to the left hasn’t been removed
> * On a 320px viewport, the page explorer scrolls horizontally due not taking over the whole screen anymore. I think we’d want no horizontal scrolling
> * In sub-menus, the menu icons don’t seem to have enough contrast against their background. We’d need a minimum of 3:1. Looks like this is because the icon has an opacity of `0.15`, on top of us changing the color to a transparent white.
> 
> And more minor things, where I might have it wrong:
> 
> * I’d expect the locale selector in the page explorer should keep having the same background as the rest of the explorer header
> * I’d expect the colors of the icons in the page explorer should be more consistent
> 
> Summary GIF:
> 
> ![sidebar-styles](https://user-images.githubusercontent.com/877585/158814989-50530b51-a6fe-424e-af5d-e66c85531764.gif)



> The end result looks great @stevedya! There are a few issues with the implementation, and a few more refactorings to do, but we’re not far off.
> 
> As far as the UI, I could spot the following issues:
> 
> * On mobile viewports, the spacing around the account toggle is missing
> * On mobile viewports, the hamburger button isn’t aligned with the close button
> * On mobile viewports, the space for the "close" button is still there even though the button has been moved to the right side of the screen
> * I can’t see the "Close" button on mobile on some pages – it’s there on `/admin/`, but not on `/admin/pages/60/`.
> * On pages like /admin/pages/60/edit/, we’ve moved the hamburger button to the right, but the extra space needed to the left hasn’t been removed
> * On a 320px viewport, the page explorer scrolls horizontally due not taking over the whole screen anymore. I think we’d want no horizontal scrolling
> * In sub-menus, the menu icons don’t seem to have enough contrast against their background. We’d need a minimum of 3:1. Looks like this is because the icon has an opacity of `0.15`, on top of us changing the color to a transparent white.
> 
> And more minor things, where I might have it wrong:
> 
> * I’d expect the locale selector in the page explorer should keep having the same background as the rest of the explorer header
> * I’d expect the colors of the icons in the page explorer should be more consistent
> 
> Summary GIF:
> 
> ![sidebar-styles](https://user-images.githubusercontent.com/877585/158814989-50530b51-a6fe-424e-af5d-e66c85531764.gif)

I believe I've fixed these mobile issues now. 
- Put the expand on the left for mobile viewports, so it will be similar to where it was before and the close button will be in the same spot of the nav as the expand
- Ensured the page navigator is set to full width on mobile. 
- Ensured the icon will be showing for the home page and back buttons to the left of a page explorer header (on mobile) 
- Allowed title / icons to flex and wrap so that text can fit along with any labels for locale/page status
- Adjusted menu icon colours by default.

(I should note ben does have plans to give the page explorer a proper reskin in the future however this is a temporary solution to match the  new sidebar styling)

Here's a screenshot at 320 on Chrome 99 (for reference)
<img width="398" alt="Screen Shot 2022-03-22 at 3 24 45 PM" src="https://user-images.githubusercontent.com/25041665/159589081-aeb593ef-9fc9-4925-bb85-ed5943978008.png">

stevedya commented on client/src/components/Sidebar/modules/WagtailBranding.tsx:
```diff
@@ -92,42 +92,21 @@ const WagtailBranding: React.FunctionComponent<WagtailBrandingProps> = ({
-        <img
-          className="sidebar-wagtail-branding__icon"
-          data-part="body"
-          src={images.desktopLogoBody}
```
Good suggestion, i've done this now. I also cleaned up any unused styles that I could see 🙂  hope thats okay. (Makes this PR a little bigger unfortunately)

stevedya commented on wagtail/admin/static_src/wagtailadmin/images/wagtail-logo.svg:
```diff
@@ -1,32 +1,32 @@
-		c-12.667,3.126-23.322,5.26-30.779,6.738C131.448,355.869,127.182,356.69,127.182,356.69"/>
-</g>
-</svg>
+<?xml version="1.0" encoding="utf-8"?>
```
Hmm it seems to have changed the line separators from CRLF to LF. Might need your help to revert this haha i'm so confused

thibaudcolas commented on client/tailwind.config.js:
```diff
@@ -55,6 +55,8 @@ module.exports = {
       LinkText: 'LinkText',
       ButtonText: 'ButtonText',
+      white: '#FFF',
+      black: '#000',
```
We already have white and black colors defined so I assume those two are useless?

thibaudcolas reviewed (approved):
Thank you @stevedya, all looking good now with the exception of:

> I’d expect the locale selector in the page explorer should keep having the same background as the rest of the explorer header

For me it’s still a different background. I don’t want your changes to risk further conflicts anyway so will merge this now – please revisit this later.
